tests/ui/join_absolute_paths.rs

```diff
@@ -0,0 +1,25 @@
+#![allow(unused)]
+#![warn(clippy::join_absolute_paths)]
+use std::path::Path;
+
+fn main() {
+    // should be linted
+    let path = Path::new("/bin");
+    path.join("/sh");
+    println!("{}", path.display());
+
+    //should be linted
+    let path = Path::new("C:\\Users");
+    path.join("\\user");
+    println!("{}", path.display());
+
+    // should not be linted
+    let path: &[&str] = &["/bin"];
+    path.join("/sh");
+    println!("{:?}", path);
+
+    //should not be linted
+    let path = Path::new("/bin");
+    path.join("sh");
+    println!("{}", path.display());
```

Comment:
These `println!`s should probably be removed. They're a bit misleading as `join` takes `&self`, not `&mut self`.

Reply:
That makes sense. Not everyone would be using absolute paths when calling join to get this behavior.